monroe-experiments/usr/bin/monroe-experiments

```diff
@@ -282,6 +282,30 @@ if [ -z "$(docker ps --no-trunc|grep base-$URL_MPLANE)" ]; then
   logger -t monroe-experiments "started mplane container.";
 fi
 
+
+### NEAT PROXY ###########################################################################
+
+URL_NEAT_PROXY="trozensztrauch/neat-proxy"
+CID_PROXY=$(docker ps --no-trunc | grep $URL_NEAT_PROXY | awk '{print $1}' | head -n 1)
+echo "TORO: neat-proxy: ${CID_PROXY}"
+
+# Configure and start neat-proxy container if not running
```

Comment:
Is this supposed to be continuously running, not just when a neat experiment is active?

Reply:
I guess we could boot it on demand if we think that is more desirable but it is designed to run all the time. 

It is the fwmark (https://github.com/MONROE-PROJECT/Scheduler/blob/8b981dd6ddf4a857de496ef4fe7138c127a7f200/files/usr/bin/container-start.sh#L141) that controls if traffic goes thorugh this container or directly out on the network interface.

Reply:
Should the firewall rules then also be continuously active, or do they interfere with normal operation?

Reply:
The fwmark should only be active when running a neat experiment (it will set a mark that redirect traffic to a specific routing table (100) that is handled by the container. AFAIK I must add as I have not written the container myself.

Reply:
Ok, then it's good to be a bit paranoid about cleaning these rules. ![isn't it?](https://media.giphy.com/media/11fUmthmh836Q8/giphy.gif)